src/neuroconv/tools/nwb_helpers/_dataset_configuration.py

```diff
@@ -129,16 +131,15 @@ def get_default_dataset_io_configurations(
                 )
 
                 yield dataset_io_configuration
-        else:
-            # Primarily for TimeSeries, but also any extended class that has 'data' or 'timestamps'
-            # The most common example of this is ndx-events Events/LabeledEvents types
-            time_series = neurodata_object  # For readability
-
-            for dataset_name in ("data", "timestamps"):
-                if dataset_name not in time_series.fields:  # The 'timestamps' field is optional
+        elif isinstance(neurodata_object, NWBContainer) and any(
+            known_dataset_field in neurodata_object.fields for known_dataset_field in known_dataset_fields
+        ):
+            for known_dataset_field in known_dataset_fields:
+                # Skip optional fields that aren't present
+                if known_dataset_field not in neurodata_object.fields:
```

Comment:
Either this line or the `any` check above is redundant I think.

Reply:
Fair. Removed